src/Npgsql/PublicAPI.Unshipped.txt

```diff
@@ -38,6 +40,9 @@ Npgsql.NpgsqlTracingOptions.ProvideSpanNameForBatch.get -> System.Func<Npgsql.Np
 Npgsql.NpgsqlTracingOptions.ProvideSpanNameForBatch.set -> void
 Npgsql.NpgsqlTracingOptions.ProvideSpanNameForCommand.get -> System.Func<Npgsql.NpgsqlCommand!, string?>?
 Npgsql.NpgsqlTracingOptions.ProvideSpanNameForCommand.set -> void
+Npgsql.NpgsqlTypeLoadingOptionsBuilder
+Npgsql.NpgsqlTypeLoadingOptionsBuilder.EnableTableCompositesLoading(bool enable = true) -> Npgsql.NpgsqlTypeLoadingOptionsBuilder!
+Npgsql.NpgsqlTypeLoadingOptionsBuilder.SetServerCompatibilityMode(Npgsql.ServerCompatibilityMode serverCompatibilityMode = Npgsql.ServerCompatibilityMode.None) -> Npgsql.NpgsqlTypeLoadingOptionsBuilder!
```

Comment:
I think it's a problem for "server compatibility mode" to be considered a type loading option.. NoTypeLoading is indeed a type loading thing, but Redshift also affects other things (specifically, Redshift doesn't support UNLISTEN so we exclude that from the pooling reset when specified...

I'd suggest flattening NoTypeLoading to be a simple thing directly on NpgsqlTypeLoadingOptionsBuilder (EnableTypeLoading() defaulting to true), rather than an enum. Redshift specifically we should ideally just get rid off, by detecting Redshift ourself - this should be possible by sending a `SELECT version()` on physical startup and searching for Redshift in there...

Reply:
Yeah I agree if that's possible. Even though the enum is broadly/badly named all its cases do only affect type loading today.

Reply:
Well that's not true for Redshift, which does affect UNLISTEN no?